tsdb/index/tsi1/partition.go

```diff
@@ -389,7 +388,7 @@ func (p *Partition) Close() error {
 	var err error
 	for _, f := range p.fileSet.files {
 		if localErr := f.Close(); localErr != nil {
-			err = localErr
+			err = errors.Join(err, localErr)
```

Comment:
`errors.Join` ignores nil errors, so no test for not nil is necessary. The code as it is written will make nested errors; the best thing to do is use a slice.  

See [here](https://github.com/influxdata/influxdb/blob/23008e5286d0161a6add86e5676ee633871c1d45/query/iterator.gen.go#L161C1-L175C1) for an example

Reply:
Done 👍